### ISSUE TITLE
Installation script's RAM requirement ignores available swap

### ISSUE THREAD

**Issue by arvoelke** [EOSIO/eos#2992]
It would be nice to do some EOS development on my chromebook (running Ubuntu 16.04), which has `4 GB` of physical memory and `4 GB` of swap:

```
.../Crypto/eos$ free -m
              total        used        free      shared  buff/cache   available
Mem:           3895        1949         896         155        1048        1373
Swap:          3999           0        3999
```

However the build script only seems to pick up on the physical RAM (the first row), without considering the available swap (the second row).

By the way, note [the documentation](https://github.com/EOSIO/eos/wiki/Local-Environment) says `8 GB` is needed, but the build script says `7 GB`:
> Your system must have **7 or more** Gigabytes of physical memory installed.

Related to PR #2338. Full output:

```
	Beginning build version: 1.2
	Sat May 12 19:31:34 UTC 2018
	User: arvoelke
	git head id: 5b62c736e9c91c3e5cf80e622507ff22d2046e83
	Current branch: master

	ARCHITECTURE: Linux

	OS name: Ubuntu
	OS Version: 16.04
	CPU speed: 2249.100Mhz
	CPU cores: 2
	Physical Memory: 3895 Mgb
	Disk install: /dev/mmcblk1p1
	Disk space total: 117G
	Disk space available: 101G
	Your system must have 7 or more Gigabytes of physical memory installed.
	Exiting now.
```

**Comment by bytemaster** [comment]
The build process will be painfully slow if you rely on swap.

**Comment by arvoelke** [comment]
Thanks for the reply.

> The build process will be painfully slow if you rely on swap.

I don't mind. I'm willing to pay the price. :) I bypassed the check in `scripts/eosio_build_ubuntu.sh` and it compiled fine in a couple of hours. The unit tests and single node testnet are working. I was watching the memory usage during compilation and it didn't appear to chew into my swap at all (the peak usage that I saw was around 25% of 8 GB coming from two processes combined), but I wasn't watching closely the entire time.

I have a hunch that the actual memory requirement may depend on the number of available cores (only `2` in my case). It appears the `build_ubuntu` script is calling `make -j${JOBS}` where `JOBS = min(CPU_CORE, MEM_GIG)` is the number of parallel processes, and `MEM_GIG = MEM_MEG / 2000` (yes, `2000`). Perhaps if `CPU_CORE < MEM_GIG`, the actual memory requirement becomes less than `8 GB`? As an example, supposing you have `8 GB` of physical memory, but only `2` cores, then `JOBS = min(2, 4)` and so only half as many jobs are run in parallel. In turn, the actual memory requirement might also be reduced in half. Hopefully this information can help others who want to try development using a lightweight Ubuntu client.
